src/test/java/com/techsophy/tsf/services/gateway/filter/MultitenancyFilterTest.java

```diff
@@ -0,0 +1,82 @@
+        Mockito.when(serverHttpRequest.mutate()).thenReturn(builder);
+        Mockito.when(builder.header(anyString(),anyString())).thenReturn(builder);
+        Mockito.when(builder.build()).thenReturn(serverHttpRequest);
+        Mockito.when(serverWebExchange.getRequest()).thenReturn(serverHttpRequest);
+        Mockito.when(serverHttpRequest.getHeaders()).thenReturn(httpHeaders);
+        Mockito.when(serverWebExchange.mutate()).thenReturn(serverWebExchangeBuilder);
+        Mockito.when(serverWebExchangeBuilder.request((ServerHttpRequest) any())).thenReturn(serverWebExchangeBuilder);
+        Mockito.when(serverWebExchangeBuilder.build()).thenReturn(serverWebExchange);
+        Mockito.when(gatewayFilterChain.filter(any())).thenReturn(Mono.empty());
+        Mockito.when(serverWebExchange.getResponse()).thenReturn(serverHttpResponse);
+        Mockito.when(serverHttpResponse.getHeaders()).thenReturn(httpHeaders);
+        MultiTenancyFilter multiTenancyFilter=new MultiTenancyFilter();
+        Mono<Void> result=multiTenancyFilter.filter(serverWebExchange,gatewayFilterChain);
+        StepVerifier.create(result)
```

Comment:
how have you verified that the outgoing request has the headers as intended.
Also what about response?

Reply:
yes verifying the response has the headers as intended